lib/scenic/adapters/postgres.rb

```diff
@@ -155,6 +170,10 @@ def try_index_create(index)
         connection.execute("ROLLBACK TO SAVEPOINT #{index.index_name}")
         "index '#{index.index_name}' on '#{index.object_name}' is no longer valid and has been dropped."
       end
+
+      def raise_unless_materialized_views_supported
+        raise Scenic::Postgres::MaterializedViewsNotSupportedError unless connection.supports_materialized_views?
```

Comment:
Additionally, if you move the files into the `Postgres` class like I had suggested above, this can just be `raise MaterializedViewsNotSupportedError`